doc/design/optimizer.md

```diff
@@ -0,0 +1,85 @@
+## Optimizer Design
+In deeplearning system, `Optimizer` is used to optimize(minimize) loss thow updating a list of parameters. 
+
+### A typical training process:
+
+1. run forward to calculate activation using data and parameter.
+1. run backward to calculate the gradient of activation and parameter using cost, activation, and parameter.
+1. run optimize operators to apply/update the gradient to the corresponding parameter.
+
+### Python Interface to describe the training process
+
+1. User write code to describe the network:
+
+```python
+images = layer.data("images")
+labels = layer.data("labels")
+w1 = pd.var("w1")
+hidden = layer.fc(images, W=w1)
+cost = layer.mse(hidden, labels)
+```
+
+the code above will generate forward operators in [block](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/block.md).
+
+
+2. User create a Optimizer and set parameter list that it need to update.
+
+```python
```

Comment:
Correct code snippet indentation in the Markdown doc.

Reply:
done